### ISSUE TITLE
Excluding a class with annotatedBy doesn't exclude inner classes / companion

### ISSUE THREAD

**Issue by NinoDLC** [Kotlin/kotlinx-kover#331]
**Describe the bug**
Using the new 0.7 API `annotatedBy(String...)` in the `excludes` lambda doesn't exclude the inner classes / companion of excluded matching classes

**Expected behavior**
If I use this kind of configuation:
```
koverAndroid {
    common {
        filters {
            excludes {
                annotatedBy(
                    "androidx.room.Database",
                )
            }
        }
    }
}
```
I should not get anything related to my class `AppDatabase` in my kover report.

**Actual behavior**
![image](https://user-images.githubusercontent.com/11870220/229130919-baac33f4-d26a-4e12-849c-8be1e5e7abdb.png)
The `AppDatabase$Companion` is not excluded even if I exclude the `AppDatabase` because it is annotated by `androidx.room.Database`

NB: `AppDatabase` itself is correctly excluded. 

**Reproducer**
https://github.com/NinoDLC/OpenClassrooms_P5_TodoK_Example/compare/master...kover_bug_example_annotatedby_inner_classes_companion

**Environment**
 - Kover Gradle Plugin version: 0.7.0-Alpha
 - Gradle version: 7.5
 - AGP: 7.4.2
 - Kotlin project type: Kotlin/Android

**Comment by shanshin** [comment]
Hi,
indeed, the companion class should also be excluded.

However, it is ambiguous about nested and internal classes - these are independent classes that can be used independently of the containing class. Therefore, nested classes should not be excluded in all cases.

**Comment by NinoDLC** [comment]
Just throwing some ideas but maybe a *private* inner class should be excluded / included the same way the parent class is excluded / included. Since the private inner class can only be used by the parent class. 

In other cases, it's ambiguous as you said.

**Comment by shanshin** [comment]
> Since the private inner class can only be used by the parent class.

Unfortunately, this is not always the case, this inner class can implement interface, and an instance of the class can be returned from the factory with the type of this interface.

In this case, there may be a need to evaluating coverage of such an inner class.

**Comment by shanshin** [comment]
Inner classes shouldn't be excluded by annotation on containing class.

The companion should be excluded - now it's a bug, see #431.

**Comment by shanshin** [comment]
Closed as a duplicate of #431
